src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -86,40 +89,84 @@ class JooqBurdenEstimateRepository(
                                   outcomeLookup: Map<String, Int>, cohortSizeId: Int,
                                   expectedDisease: String)
     {
-        val records = estimates.flatMap { estimate ->
-            if (estimate.disease != expectedDisease)
-            {
-                throw InconsistentDataError("Provided estimate lists disease as '${estimate.disease}' but scenario is for disease '$expectedDisease'")
+        val countries = getAllCountryIds()
+
+        // The only foreign keys are:
+        // * burden_estimate_set, which is the same for every row, and it's the one we just created and know exists
+        // * country, which we check below, per row of the CSV (and each row represents multiple rows in the database
+        //   so this is an effort saving).
+        // * burden_outcome, which we check below (currently we check for every row, but given these are set in the
+        //   columns and don't vary by row this could be made more efficient)
+        dsl.withoutCheckingForeignKeyConstraints(BURDEN_ESTIMATE) {
+
+            // Currently we write everything to a byte array in memory and then later feed that stream to CopyManager.
+            // Obviously this could be made more performant by chaining those two together somehow.
+            val data = ByteArrayOutputStream().use { stream ->
+                PostgresCopyWriter(stream).use { writer ->
+                    for (estimate in estimates)
+                    {
+                        if (estimate.disease != expectedDisease)
+                        {
+                            throw InconsistentDataError("Provided estimate lists disease as '${estimate.disease}' but scenario is for disease '$expectedDisease'")
+                        }
+                        if (estimate.country !in countries)
+                        {
+                            throw UnknownObjectError(estimate.country, "country")
+                        }
+
+                        writer.writeRecord(newBurdenEstimateRecord(setId, estimate, cohortSizeId, estimate.cohortSize))
```

Comment:
`writeRecord` is a little ambiguous; this is writing a line to a text file right? maybe `writeLine` or `writeValues` or something similar?

Reply:
It is writing to a text file, but each line of the text file still corresponds to inserting a row into the database, so I felt like here (and with your comment below) I didn't want to go as vague as "values". It's still a record, although it's not a Jooq record. Maybe `writeRow` and `newBurdenEstimateRow`?

Reply:
Those sound good to me :+1:

Reply:
Made the changes.